### ISSUE TITLE
Redis native support

### ISSUE THREAD

**Issue by aldettinger** [apache/camel-quarkus#2107]

**Comment by aldettinger** [comment]
camel-redis uses [Reddisson](https://github.com/redisson/redisson) under the hood, it's not quarkus native friendly.
It would need a solution to manage or disable [netty native transport](https://netty.io/wiki/native-transports.html).

**Comment by aldettinger** [comment]
The quarkus community manages redis with [vertx-redis-client](https://github.com/quarkusio/quarkus/issues/4453#issuecomment-542042896). Therefore, it's worth having a look into using vertx-redis-client in camel-redis.

**Comment by davsclaus** [comment]
Yeah vertx components are good to use, they are tiny, fast, reactive and support native etc.

**Comment by aldettinger** [comment]
Redisson offers an abstraction level for transactions and distributed collections/locks while vertx-redis-client does not.
We would therefore be left with implementing that with redis primitives on top of vertx-redis-client.
At the end of the day, I think it would void the foreseen benefits of aligning on quarkus.

On the other hand, redisson depends upon netty 4 where we already have had success with native support.
So, I'm going to explore this path.

**Comment by aldettinger** [comment]
Work in progress in this branch https://github.com/aldettinger/camel-quarkus/commits/CAMEL-QUARKUS-2107

**Comment by ineednousername** [comment]
@aldettinger did you already have a look at this project? https://github.com/redisson/redisson/tree/master/redisson-quarkus

**Comment by aldettinger** [comment]
@ineednousername, I've not yet had a look to redisson-quarkus but will for sure. Many thanks for the pointer to this project :+1:
